pageserve/app.py

```diff
@@ -104,28 +110,27 @@ def add_event():
 
 def get_posts():
     """Gets all posts from posts service."""
-    # TODO: integrate with posts service to pull post info from database
-    posts = [{'post_id': '1',
-              'event_id': '0',
-              'type': 'text',
-              'author': 'admin',
-              'created_at': '7-9-2019',
-              'text': 'this will be a fun event!',
-              },
-             {'post_id': '2',
-              'event_id': '0',
-              'type': 'image',
-              'author': 'admin',
-              'created_at': '7-9-2019',
-              'text': 'abcdefghi',
-              }]
-    return parse_posts(posts)
-
-
-def parse_posts(posts):
-    # TODO(cmei4444): implement parsing on posts pulled from posts service in
-    # a format for web display
-    return posts
+    url = app.config['POSTS_ENDPOINT']
+    r = requests.get(url, params={})
+    if r.status_code == 200:
+        return parse_posts(r.json())
+    raise RuntimeError("Error in retrieving posts.")
+
+
+def parse_posts(posts_dict):
+    """Parses response from posts service to be used in HTML templates.
+
+    Args:
+        events_dict: JSON returned by posts service, includes:
+            posts (list): list of posts
+            num_posts (int): number of posts returned
+
+    Returns:
+        list: parsed list of posts.
+    """
+    # TODO(cmei4444): implement parsing on posts - timestamps are formatted
+    # unreadably currently
```

Comment:
It sounds like you've done more research into how `datetime`s work in the db than I have, what kind of parsing would be needed? Depending on what we want to do, it might be simple to do the parsing in posts and events before we upload stuff to the database and just store them as strings. If you try out `str(datetime.datetime.now())` it actually returns a fairly readable time string.

Reply:
I think I actually figured it out - the events service at least is giving Unix times in milliseconds, which are pretty easy to parse: 
```
ts = int(unix_time)/1000
event['created_at'] = datetime.utcfromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S')
```

In the DB datetimes are actually stored pretty readably (from looking at the MongoDB atlas site), the issue happens when you try to JSON serialize them. 
Storing them as strings would be easier for testing at this point though so it honestly might be preferable.

Reply:
I was just testing things and `datetime.datetime.utcnow().isoformat(" ", "seconds")` returns `'2019-07-31 23:52:18'`. I highly recommend we use this when creating timestamps and store the times as strings like that in the DB. If we don't care about globalization, we can drop the `utcnow()` to just `now()` for local time, but I figure it's probably good principle for a social app like this to use standard time.

Reply:
That seems good, let's go with that. I'll implement that for events in another PR.

Reply:
Cool, I'll implement it for posts in another PR.